t2.go

```diff
@@ -0,0 +1,88 @@
+package ftdc
+
+import (
+	"context"
+	"io"
+	"log"
+	"math"
+
+	"github.com/evergreen-ci/birch"
+	"github.com/pkg/errors"
+)
+
+const SECOND_MS int64 = 1000
+const MAX_SAMPLES = 300
+
+func TranslateGenny(ctx context.Context, iter *ChunkIterator, output io.Writer, actorOpName string) error {
+	collector := NewStreamingCollector(MAX_SAMPLES, output)
+	var err error
+	defer func() {
+		err = FlushCollector(collector, output)
+	}()
+
+	currentSecond := int64(0)
+	endOfSecondIdx := -1
+	prevChunk := iter.Chunk()
+
+	for iter.Next() {
+		if err = ctx.Err(); err != nil {
+			return errors.New("operation aborted")
+		}
+		currChunk := iter.Chunk()
+		elems := make([]*birch.Element, 0)
+		var startTime *birch.Element
+
+		timestamp := currChunk.Metrics[0]
+
+		for idx, ts := range timestamp.Values {
+			if math.Ceil(float64(ts)/float64(SECOND_MS)) != float64(currentSecond) && endOfSecondIdx > -1 {
+
+				chunk := currChunk
+				if endOfSecondIdx == len(prevChunk.Metrics[0].Values)-1 {
+					chunk = prevChunk
+				}
+
+				for _, metric := range chunk.Metrics {
+					switch name := metric.Key(); name {
+					case "ts":
+						startTime = birch.EC.DateTime("start", currentSecond*SECOND_MS)
+					case "counters.n":
+						elems = append(elems, birch.EC.Int64("n", metric.Values[endOfSecondIdx]))
```

Comment:
Is it possible for the `Values` slices to have different lengths across different metrics within the same chunk? In this case we could panic.